doc/design/python/user_interface.md

```diff
@@ -0,0 +1,370 @@
+
+Similarlly, the function of `pd.Block` in Paddle is to enable groups of operators to be treated as if they were one operator to make `if_else_op` or RNNOp's declaration simpler and Python's `with` statement is used to make the codes look much like a block.
+
+For example, when defining a `RNNOp`, we can use `pd.Block` to help configure a step network:
+
+```python
+v = some_op()
+m_boot = some_op()
+
+W = pd.Variable(shape=[20, 20])
+U = pd.Variable(shape=[20, 20])
+
+rnn0 = RNNOp()
+with rnn0.stepnet() as net:
+    # declare stepnet's inputs
+    x = net.add_input(v)
+    # declare memories
+    h = net.add_memory(m_boot)
+
+    fc_out = pd.matmul(W, x)
+    hidden_out = pd.matmul(U, h)
+    sum = pd.add_two(fc_out, hidden_out)
+    act = pd.sigmoid(sum)
+
+    # declare stepnet's outputs
+    net.add_output(act, hidden_out)
+
+acts, hs = rnn0()
+```
+
+The operators inside the `with`-statement defines the rnn's step network, 
+and will be put into a `pd.Block`.
+
+another example is the definition of `if_else_op`:
+
+```python
+# v0 is a output of some_op
+v0 = some_op()
+v1 = some_op()
+
+ifelseop = pd.if_else_op()
+with ifelseop.true_block() as net:
+    x0, x1 = net.add_input(v0, v1)
+    
+    y = pd.fc(x)
+    z = pd.add_two(x1, y)
+    
+    net.add_output(z)
+
+with ifelseop.false_block() as net:
+    x0, x1 = net.add_input(v0, v1)
+    
+    y = pd.add_two(x0, x1)
+    
+    net.add_output(y)
+    
+# output of ifelseop
+out = ifelseop()
+```
+
+In most cases, user need not to create a `pd.Block` directly, but it is the basis of a Paddle program:
+
+- user's program is stored in `pd.Block`
+- when run the codes, just execute the corresponding `pd.Block`
+
+A `pd.Block` can has its own namespace, which makes it possible to hide the local variables from block, for example:
+
+```python
+W = pd.Variable(shape=[20, 20])
+
+# a and b are outputs of some_op
+a = some_op()
+b = some_op()
+
+with pd.Block('namespace0'):
+    # W is a local variable and has its own value
+    W = pd.Variable(shape=[20, 20])
+    x = pd.matmul(W, a)
+    y = x + b
+    
+with pd.Block('namespace1'):
+    # W is the global variable
+    z = pd.matmul(W, a)
+    
+# g use local variables in both namespace0 and namespace1
+g = pd.add_two(y, z)
+```
+
+### Op (short for Operator)
+`Op` defines basic operation unit of optimized computation graph in Paddle, one `Op` has several input and output variables, and some attributes.
+
+Take `pd.matmul` for example, one can use it like this
+
+```python
+out = pd.matmul(a, b)
+```
+which means that a operator `pd.matmul` takes two variables `a` and `b` for input, 
+and return a variable `out`.
+
+### Layer
+`Layer` defines a more complex operation which may combines several `Op`s, its usage is the same with `Op`.
+
+Take `pd.fc` for example, one can use it like this
+```python
+out = pd.fc(in, param_names=['W'])
```

Comment:
1. Is parameter `w` defined outside of fc layer or inside the layer?
2. If a layer has more than one parameter, how can we know the name correspondence?

Reply:
currently, maybe layer create parameters better. the big difference between a op and a layer.